### ISSUE TITLE
Accessibility problems in `chosen` dropdowns

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-admin#1955]
An accessibility review has indicated there are some potential accessibility problems with `chosen` dropdowns.

Some problematic dropdowns include:

- The sort dropdown in asset admin
- The fluent locale dropdown in the left menu
- dropdowns as used in `Member` edit forms (e.g. "Interface Language" and "Groups")

It doesn't appear the normal `DropdownField` has these problems - or if it does, the tool didn't pick up on them.

### [WCAG 2.2 A F15](https://www.w3.org/TR/WCAG20-TECHS/F15) 4.1.2 Clickable controls should be keyboard accessible

Because of an `<a>` tag in the chosen markdown, tooling that looks for accessibility problems thinks the `<a>` tag needs a `tabindex=0` attribute added to it.

Note that the chosen dropdowns _do_ get focus when you tab to them, but it's not the `<a>` that gets focus. This may be a red herring - but ideally `<a>` tag shouldn't be there if it's not an interactive element.

### [WCAG 2.2 A F54](https://www.w3.org/TR/WCAG20-TECHS/F54) 2.1.1 Clickable controls should have an ARIA role

Because of an `<a>` tag in the chosen markdown, tooling that looks for accessibility problems thinks the `<a>` tag needs an aria `role` attribute such as `aria="button"` (in this case since there's no `href` attribute).

This may be a red herring - but ideally `<a>` tag shouldn't be there if it's not an interactive element - and if it is interactive, it needs to be identifiable as such, which is what the `role` does.

### [WCAG 2.2 A F68](https://www.w3.org/TR/WCAG20-TECHS/F68.html) 4.1.2 HTML form control has no accessible name

Chosen dropdowns sometimes have an element that looks like `<input class='chosen-search-input' type='text' autocomplete='off' readonly>`. Because this has no label (neither a parent `<label>` element nor any of the `title`, `aria-label`, or `aria-labelledby` attributes), it gets flagged as not having a label.

Likely chosen doesn't let you directly interact with this element and therefore this may be a red herring.

## Note

- `chosen` is used in entwine forms, but double check react dropdowns as well just in case
- These may not be resolvable - see https://github.com/silverstripe/silverstripe-admin/issues/2027
- Because the internal `<input>` is given focus instead of the actual visible element, we can't give sensible focus-visible states to some of these (e.g. the preview mode selector)

**Comment by GuySartorelli** [comment]
There are a bunch more problems it turns out, and none of them are likely to be resolved without replacing chosen, sadly.
